src/app/backend/resource/deployment/common.go

```diff
@@ -64,3 +67,23 @@ func fromCells(cells []dataselect.DataCell) []apps.Deployment {
 	}
 	return std
 }
+
+func getStatus(list *apps.DeploymentList, rs []apps.ReplicaSet, pods []v1.Pod, events []v1.Event) common.Status {
+	info := common.Status{}
+
+	for _, deployment := range list.Items {
+		matchingPods := common.FilterDeploymentPodsByOwnerReference(deployment, rs, pods)
+		podInfo := common.GetPodInfo(deployment.Status.Replicas, deployment.Spec.Replicas, matchingPods)
+		warnings := event.GetPodsEventWarnings(events, matchingPods)
+
+		if len(warnings) > 0 {
+			info.Failed++
+		} else if podInfo.Pending > 0 {
+			info.Pending++
+		} else {
+			info.Running++
+		}
```

Comment:
Export to a function if it repeats somewhere?

Reply:
It does not